Sources/TracingMacros/TracedMacro.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2024 Apple Inc. and the Swift Distributed Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Distributed Tracing project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+@_exported import ServiceContextModule
+import Tracing
+
+#if compiler(>=6.0)
+@attached(body)
+public macro Traced(
+    _ operationName: String? = nil,
```

Comment:
I wonder if this may want to be an `TracedOperationName` like this:

```
public struct TracedOperationName: ExpressibleByStringLiteral{ 
let value: TracedOperationName

package enum TracedOperationName {
case baseName // the default "foo"
case fullName // "foo(bar:)
case string(String)
}


public static var baseName: ... // for all the above
```

WDYT?

Reply:
I was going back and forth on whether it should be implemented with `baseName` / `fullName` semantics, so this looks good to me! Should this type be in `TracingMacros`?

Reply:
Yup, right next to this `Traced` declaration would be fine

Reply:
Actually I just realized this is also made slightly harder by the macro—we want this conceptually but we only have access to the syntax and not the evaluated value. Two options here:
- Directly match looking for `.baseName` / `.fullName` / string expression, or add a bit of runtime support expanding to something like:

```swift
withSpan(.baseName.text(baseName: "foo", fullName: "foo(bar:)")) { span in ... }
```

And then doing the selection in regular code.

Reply:
Yes, it would be in TracingMacros I think. I think basename as basic is okey